larreco/ClusterFinder/ClusterAna_module.cc

```diff
@@ -115,611 +138,471 @@ namespace cluster {
+              myprt << " -> Cls " <<cls.ID();
+              auto clsHits = fmch.at(match.clsIndex[plane]);
+              myprt << " nRecoHits " << clsHits.size();
+              myprt << " nTruRecoHits " << match.clsTruHitCount[plane] << "\n";
+            } // match.clsTruHitCount[plane] > 0
+            else {
+              myprt<<" *** No Cluster match\n";
+            }
+          } // plane
+        } // match
+      } else {
+        myprt << "--> truncated print of " << matches.size() << " selected MCParticles";
       }
-      // fill histograms
-      if (isCosmic) {
-        fCREP2->Fill(ep2);
-        fCRE->Fill(eff[ep2Plane]);
-        fCRP->Fill(pur[ep2Plane]);
-        aveCREP2 += ep2;
-        numCREP2 += 1.;
-        if (fPrintLevel > 1)
-          mf::LogVerbatim("ClusterAna")
-            << ">>>CREP2 " << std::fixed << std::setprecision(2) << ep2 << " E " << eff[ep2Plane]
-            << std::setprecision(2) << " P " << pur[ep2Plane] << " P:W:T " << ep2Plane << ":"
-            << (int)clBeg[0] << ":" << (int)clBeg[1] << "-" << ep2Plane << ":" << (int)clEnd[0]
-            << ":" << (int)clEnd[1] << " PDG " << PDG << " KE " << (int)KE << " MeV";
-      } // isCosmic
-      else {
-        float wght = 1.;
-        if (fTrackWeightOption == 1) wght = KE;
-        // accumulate statistics for muons and not-muons
-        if (PDG == 13) {
-          aveNuEP2mu += ep2 * wght;
-          numNuEP2mu += wght;
-        }
-        else {
-          aveNuEP2nm += ep2 * wght;
-          numNuEP2nm += wght;
-        }
-        if (PDG == 11) {
-          fNuKE_elec->Fill(KE, wght);
-          fNuE_elec->Fill(eff[ep2Plane], wght);
-          fNuP_elec->Fill(pur[ep2Plane], wght);
-          fNuEP2_elec->Fill(ep2, wght);
-          fNuEP2_KE_elec->Fill(KE, ep2, wght);
-        }
-        else if (PDG == 13) {
-          fNuKE_muon->Fill(KE, wght);
-          fNuE_muon->Fill(eff[ep2Plane], wght);
-          fNuP_muon->Fill(pur[ep2Plane], wght);
-          fNuEP2_muon->Fill(ep2, wght);
-          fNuEP2_KE_muon->Fill(KE, ep2, wght);
-        }
-        else if (PDG == 211) {
-          fNuKE_pion->Fill(KE, wght);
-          fNuE_pion->Fill(eff[ep2Plane], wght);
-          fNuP_pion->Fill(pur[ep2Plane], wght);
-          fNuEP2_pion->Fill(ep2, wght);
-          fNuEP2_KE_pion->Fill(KE, ep2, wght);
-        }
-        else if (PDG == 321) {
-          fNuKE_kaon->Fill(KE, wght);
-          fNuE_kaon->Fill(eff[ep2Plane], wght);
-          fNuP_kaon->Fill(pur[ep2Plane], wght);
-          fNuEP2_kaon->Fill(ep2, wght);
-          fNuEP2_KE_kaon->Fill(KE, ep2, wght);
+    } // fPrintLevel > 1
+
+    // Calculate Efficiency and Purity
+    ++fEventsProcessed;
+    for(auto& match : matches) {
+      if (match.mcpi == UINT_MAX) continue;
+      auto& mcp = (*mcps)[match.mcpi];
+      float TMeV = 1000 * (mcp.E() - mcp.Mass());
+      int indx = PDGCodeIndex(mcp.PdgCode());
+      if (indx < 0 || indx > 4) continue;
```

Comment:
Better to have a named utility like:

`if (unsupported_index(indx)) continue;`